app/jinja_filters.py

```diff
@@ -52,6 +52,29 @@ def format_percentage(value):
     return '{}%'.format(value)
 
 
+@blueprint.app_template_filter()
+def format_address_list(user_entered_address=None, metadata_address=None):
+    """
+        This function format_address_list accepts two addresses, a user submitted address and a metadata address
+        If all the items in address list are 'Undefined' (nothing in answer_store) then we retrieve the metadata.
+
+        :param user_entered_address user entered address
+        :param metadata_address is metadata address
+        :return: the value of the address to be piped
+    """
+    if all(isinstance(a, Undefined) for a in user_entered_address):
+        address = metadata_address
+    else:
+        address = user_entered_address
+
+    address_list = Markup('<br />').join([i for i in address if i])
+
+    if not address_list:
+        raise Exception('No valid address passed to format_address_list filter')
```

Comment:
I wonder if exception it a bit harsh?

Reply:
That's what I'm thinking, if it's required to have at least one of these, then exception would be right, but if not then I'm not too sure, possible to redo this so that if the `user_entered_address` is empty then it defaults back to the `metadata_address` I think that way may be better?

Reply:
You need to check the behaviour with Donna

Reply:
Did you resolve this @dcdarrell9 ?